src/Management/src/EndpointCore/SpringBootAdminClient/BootAdminAppBuilderExtensions.cs

```diff
@@ -0,0 +1,89 @@
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Steeltoe.Management.Endpoint.Health;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+namespace Steeltoe.Management.Endpoint.SpringBootAdminClient
+{
+    public static class BootAdminAppBuilderExtensions
+    {
+        private static RegistrationResult registrationResult;
+
+        internal static RegistrationResult RegistrationResult { get => registrationResult; }
+
+        public static void RegisterSpringBootAdmin(this IApplicationBuilder builder, IConfiguration configuration, HttpClient testClient = null)
+        {
+            if (builder is null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (configuration is null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            var options = new BootAdminClientOptions(configuration);
+            var mgmtOptions = new ManagementEndpointOptions(configuration);
+            var healthOptions = new HealthEndpointOptions(configuration);
+            var basePath = options.BasePath;
+
+            var app = new Application()
+            {
+                Name = options.ApplicationName ?? "Steeltoe",
+                HealthUrl = new Uri($"{basePath}{mgmtOptions.Path}/{healthOptions.Path}"),
+                ManagementUrl = new Uri($"{basePath}{mgmtOptions.Path}"),
+                ServiceUrl = new Uri($"{basePath}/"),
+                Metadata = new Metadata() { Startup = DateTime.Now }
+            };
+            var lifetime = builder.ApplicationServices.GetService<IHostApplicationLifetime>();
+            lifetime.ApplicationStarted.Register(() =>
+            {
+                var httpClient = testClient ?? new HttpClient();
+
+                var content = JsonConvert.SerializeObject(app);
+                var buffer = System.Text.Encoding.UTF8.GetBytes(content);
+                var byteContent = new ByteArrayContent(buffer);
+                byteContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/json");
+                var result = httpClient.PostAsync($"{options.Url}/instances", byteContent).Result;
+                if (result.IsSuccessStatusCode)
+                {
+                    var task = result.Content.ReadAsStringAsync();
+                    registrationResult = JsonConvert.DeserializeObject<RegistrationResult>(task.Result);
+                }
+            });
+
+            lifetime.ApplicationStopped.Register(() =>
+            {
+                if (RegistrationResult == null || string.IsNullOrEmpty(RegistrationResult.Id))
+                {
+                    return;
+                }
+
+                var httpClient = testClient ?? new HttpClient();
```

Comment:
Any way to use `HttpClientFactory` or at least the static method from Steeltoe.Common here?